logging/logmanager/plugin_impl_log_manager.go

```diff
@@ -216,3 +234,91 @@ func stringToLogLevel(level string) logging.LogLevel {
 
 	return logging.InfoLevel
 }
+
+// list of supported hook services to send errors to remote syslog server
+const (
+	HookSysLog   = "syslog"
+	HookLogStash = "logstash"
+	HookFluent   = "fluent"
+)
+
+// HookConfig contains configuration of hook services
+type HookConfig struct {
+	Protocol string
+	Address  string
+	Port     int
+	Levels   []string
+}
+
+// CommonHook implements that Hook with own level definition
+type CommonHook struct {
+	logrus.Hook
+	levels      []logrus.Level
+}
+
+// Levels overrides implementation from embedded interface
+func (cH *CommonHook) Levels() []logrus.Level {
+	return cH.levels
+}
+
+// AddHook adds hook for existing loggers and store it into registy for late use
+func (lm *Plugin) AddHook(hookName string, hookConfig HookConfig) error {
+	var cHook *CommonHook
+	var lgHook logrus.Hook
+	var err error
+
+	switch hookName {
+	case HookSysLog:
+		var address = hookConfig.Address
+		if hookConfig.Address != "" {
+			address = address + ":" + strconv.Itoa(hookConfig.Port)
+		}
+		lgHook, err = lgSyslog.NewSyslogHook(
+			hookConfig.Protocol,
+			address,
+			syslog.LOG_INFO,
+			"")
+	case HookLogStash:
+		lgHook, err = logrustash.NewHook(
+			hookConfig.Protocol,
+			hookConfig.Address+":"+strconv.Itoa(hookConfig.Port),
+			"vpp-agent")
+	case HookFluent:
+		lgHook, err = logrus_fluent.NewWithConfig(logrus_fluent.Config{
+			Host: hookConfig.Address,
+			Port: hookConfig.Port,
+		})
+	default:
+		return fmt.Errorf("unsupported hook")
+	}
+	// create hook
+	cHook = &CommonHook{
```

Comment:
You don't have to define this variable at the top of the function, you can just use `cHook := &CommonHook{`.